HubSpot module (#762)
crowdbotics/modules: modules/django-hubspot/django_hubspot/hubspot/README.md

## Crowdbotics HubSpot Component - Backend

This module contains all needed resources to get the HubSpot component for React
Native mobile client.

## HubSpot Setup

> First thing to make this module work, is create an app on the HubSpot.
> Here is the process to create an app on hubspot.

Reviewer: ```suggestion
In order to configure the module, the first thing you should do is to create an app on Hubspot. Here's the steps detailing that process:
```

1. SignIn to the [HubSpot](https://www.hubspot.com/).
2. Create a [developer account](https://developers.hubspot.com/) on HubSpot.

> 3. You will be ask to complete the few steps by entering your details.

Reviewer: ```suggestion
3. You will be asked to complete a few steps by entering your details.
```

> 4. Once the account is created user will be move to create app page.

Reviewer: ```suggestion
4. Once the account is created you will be redirected to the app creation page.
```

| `/modules/hubspot/service/deals/single/` | `{id}` | Takes object containing `id` of the deal going to be retrieved.|
| `/modules/hubspot/service/tickets/create/` | `{amount, closedate, dealname, hubspot_owner_id, pipeline, dealstage}` | Create a ticket with the given properties and return a copy of the object, including the ID|
| `/modules/hubspot/service/tickets/list/` | - | The tickets endpoint retrieves all tickets data from HubSpot.|
| `/modules/hubspot/service/tickets/single/` | `{id}` | Takes object containing `id` of the ticket going to be retrieved.|
| `/modules/hubspot/service/tickets/remove/` | `{id}` | Takes object containing `id` of the ticket going to be deleted. Moves an Object identified by `id` to the recycling bin.|
| `/modules/hubspot/service/ticket/associations/create/` | `{ticketId, toObjectType, toObjectId,param[{associationCategory, associationTypeId}]}` | Associate a ticket with others CRM objects.|
| `/modules/hubspot/service/ticket/associations/list/` | `{ticketId, toObjectType}` | Reterive a ticket associated with other CRM objects.|

> | `/modules/hubspot/service/contact/deals/list/` | `{contactId}` | Reterive a contact associated with deals.|

Reviewer: ```suggestion
| `/modules/hubspot/service/contact/deals/list/` | `{contactId}` | Retrieve a contact associated with deals.|
```

> | `/modules/hubspot/service/meeting/contacts/list/` | `{meetingId}` | Reterive a meeting associated with contacts.|

Reviewer: ```suggestion
| `/modules/hubspot/service/meeting/contacts/list/` | `{meetingId}` | Retrieve a meeting associated with contacts.|
```
